### PR TITLE
Hotfix dm button

### PR DESCRIPTION
**Pull request by EandrewJones**
Implements a number of fixes

## Fixes

- Add prevent default on enter hook to all forms
- Make posts/replies/comments fetch most up-to-date author photo on mount rather than the one statically linked to the post (we should do this with all user info)
- Add temporary settings page that displays `UserProfileForm` modal so users can update profiles after first login
- Fixes the broken DM toggle in `UserProfileForm`
- Increases the right margin on user dropdown menu so it's no longer cut off by screen
- Fix nav links highlighting

## Persisting Bugs
- User profiles are not being created during sign up, causing the SSR error because the profile is undefined on log-in
- It appears some users aren't being created either...

### CODE REVIEW

File: components/Feed/Comments/CommentHeader.tsx [hunk 1]
```diff
@@ -1,102 +1,115 @@
...
+        db.collection('posts')
             .doc(postId)
-            .collection("comments") // Or whatever the name of the collection is
+            .collection('comments') // Or whatever the name of the collection is
             .doc(commentId)
             .delete()
             .catch((err) => {
-                console.log("Cannot delete post: ", err);
-            });
+                console.log('Cannot delete post: ', err)
```
Comment (FedericoCampe8) [inline components/Feed/Comments/CommentHeader.tsx:54]: cannot delete comment?

File: components/Feed/Comments/CommentHeader.tsx [hunk 2]
```diff
@@ -1,102 +1,115 @@
...
+        db.collection('users')
             .doc(user.uid)
             .get()
             .then((doc) => {
-                let tmp = doc.data();
-                console.log(tmp.comments);
-                delete tmp.comments[commentId];
-                doc.ref.update(tmp);
+                let tmp = doc.data()
+                console.log(tmp.comments)
```
Comment (FedericoCampe8) [inline components/Feed/Comments/CommentHeader.tsx:63]: remove

Comment (FedericoCampe8) [inline components/Feed/Comments/CommentHeader.tsx:63]: we prob want to refactor this logic after. maybe add a TODO

File: components/Feed/Post/Post.tsx [hunk 3]
```diff
@@ -1,117 +1,132 @@
...
+                    // look at listening only for changes in the children elements
+                    // to avoid issues during post deletion
                     if (isComparePost(postData)) {
                         // Add a counter of votes for each object to compare.
                         // Note: this should generally be an array of 2 objects
                         let votesCounter = new Array(
                             postData.compare.votesObjMapList.length
-                        ).fill(0);
+                        ).fill(0)
                         for (var i = 0; i < votesCounter.length; i++) {
```
Comment (FedericoCampe8) [inline components/Feed/Post/Post.tsx:62]: note: there is a ticket to refactor this with code example in the description.

**Review by FedericoCampe8 — APPROVED** [review]

**Review by FedericoCampe8 — APPROVED** [review]

**Review by FedericoCampe8 — APPROVED** [review]

**Review by FedericoCampe8 — APPROVED** [review]

**Review by FedericoCampe8 — DISMISSED** [review]

File: components/Feed/Comments/CommentHeader.tsx (hunk 1 above)
Comment (EandrewJones) [reply components/Feed/Comments/CommentHeader.tsx:54]: Fixed.

**Review by FedericoCampe8 — APPROVED** [review]

**Pull request merged**